dap-mode.el

```diff
@@ -1167,25 +1186,38 @@ after selecting configuration template."
                            (error "There is no debug provider for language %s"
                                   (or (plist-get debug-args :type) "'Not specified'")))))
 
-(defun dap-debug-edit-template (debug-args)
-  "Edit template DEBUG-ARGS."
-  (interactive (list (dap--select-template)))
-  (with-current-buffer (or (get-buffer "*DAP Templates*")
-                           (with-current-buffer (get-buffer-create "*DAP Templates*")
-                             (emacs-lisp-mode)
-                             (current-buffer)))
-    (goto-char (point-max))
-    (insert "\n\n(dap-debug `(")
-    (-let ((column (current-column))
-           ((fst snd . rst) debug-args))
-      (insert (format "%s %s" fst (prin1-to-string snd)))
-      (cl-loop for (k v) on rst by (function cddr)
-               do (progn
-                    (insert "\n")
-                    (--dotimes column (insert " "))
-                    (insert (format "%s %s" k (prin1-to-string v))))))
-    (insert "))"))
-  (pop-to-buffer "*DAP Templates*"))
+(defun dap-debug-edit-template (&optional parg debug-args)
+  "Edit registered template DEBUG-ARGS.
+When being invoked with prefix argument, poping up the prepopulated version of the template.
+Otherwise, return its original version. After registration, the new template can be used
+normally with dap-debug"
+
+  (interactive "P")
+  (let ((debug-args (dap--select-template(not parg))))
+    (progn
+      (message "debug-args: %s" debug-args)
+      (with-current-buffer (or (get-buffer "*DAP Templates*")
+			       (with-current-buffer (get-buffer-create "*DAP Templates*")
+				 (emacs-lisp-mode)
+				 (current-buffer)))
+	  (goto-char (point-max))
+	  (insert
+	   (format "\n\n(dap-register-debug-template \"%s%s\"\n"
+		   (plist-get debug-args :name)
+		   (if parg " - Copy" "")))
```

Comment:
probably this should generate unique name similar to what we do in `dap--calculate-unique-name` .

Reply:
(not to be addressed as part of this PR).

Reply:
ok.